core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -604,6 +624,11 @@ private void validateScanResults(
         return;
       }
 
+      if (notFullyScannedScanner) {
+        // If the scanner is not fully scanned, no further checks are needed
+        return;
+      }
```

Comment:
If the scanner has been not fully scanned, we don't need a further check. This is similar to the limit case:
https://github.com/scalar-labs/scalardb/blob/1717cfb7842cf674580490aad4b0c445f2211377/core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java#L622-L625